File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/mode/ModeDao.java
package ru.nikshlykov.englishwordsapp.db.mode;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ModeDao {

    @Update
    int update(List<Mode> modes);

    /*@Query("SELECT * FROM Modes WHERE IsSelected = 1")
    Mode[] getSelectedModes();*/

    @Query("SELECT * FROM Modes WHERE IsSelected = 1")
    List<Mode> newGetSelectedModes();

    @Query("SELECT * FROM Modes")
    LiveData<List<Mode>> getLiveDataModes();
}

File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/mode/ModeDao.kt
package ru.nikshlykov.englishwordsapp.db.mode

import androidx.room.Dao
import androidx.room.Query
import androidx.room.Update

@Dao
interface ModeDao {
  @Update
  suspend fun update(modes: List<Mode>): Int

  @Query("SELECT * FROM Modes WHERE isSelected = 1")
  suspend fun getSelectedModes(): List<Mode>

  @Query("SELECT * FROM Modes")
  suspend fun getAllModes(): List<Mode>
}